### ISSUE TITLE
Debugger: the stop button does not stop the app

### ISSUE THREAD

**Issue by duzenko** [flutter/flutter#64241]
## Steps to Reproduce

1. Start-debug any flutter app in Android Studio
2. Click the stop button

**Expected results:** IDE stops debugging and the app terminates

**Actual results:** IDE stops debugging but the app keeps working on the device

Command line `flutter run` and then 'q' works so it has to be an IDE/plugin issue.

<details>

Device: LG H873

  <summary>Logs</summary>

```
C:\dev\pss-mobile-android>flutter doctor -v
[√] Flutter (Channel stable-logcat-fix, v1.17.0-4.0.pre.6, on Microsoft Windows [Version 10.0.18362.959], locale ru-UA)
    • Flutter version 1.17.0-4.0.pre.6 at C:\dev\flutter
    • Framework revision bffffb8cd7 (3 months ago), 2020-06-04 11:58:54 +0300
    • Engine revision b851c71829
    • Dart version 2.8.3


[√] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at c:\tools\Android-SDK
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = c:\tools\Android-SDK
    • Java binary at: C:\tools\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Android Studio (version 4.0)
    • Android Studio at C:\tools\android-studio
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code, 32-bit edition (version 1.48.0)
    • VS Code at C:\Program Files (x86)\Microsoft VS Code
    • Flutter extension version 3.13.2

[√] Connected device (1 available)
    • LG H873 • LGH8736bb28167 • android-arm64 • Android 8.0.0 (API 26)

• No issues found!
```

</details>

**Comment by duzenko** [comment]
No, this is device-specific and thus is Flutter's issue, not plugins
It works on an emulator, but not with LG H873
https://github.com/flutter/flutter/pull/58670

**Comment by pedromassangocode** [comment]
> Command line `flutter run` and then 'q' works so it has to be an IDE/plugin issue.

Based on this statement I conclude that this definitely is a Flutter IDE/plugin issue. If it where a Flutter issue then the 'q' command wouldn’t work.

**Comment by stevemessick** [comment]
The IDE does not use the `q` command. We (plugin authors) think this is in the Flutter tool. It is likely something similar to the problem @duzenko fixed a couple months ago that prevented logcat from working on his specific device.

**Comment by pedromassangocode** [comment]
Thanks for confirmation.

**Comment by hacker1024** [comment]
Lately, I've been getting the complete opposite - my app is stopping, but IntelliJ is acting like it's not. The run button continues to show a green dot, and nothing happens when I press it again (other than going grey for a second).

Closing and opening the project seems to reset it, but upon closing, I need to press "disconnect", "cancel", and "disconnect" again.

Could this be related, or should I open a new issue in the IntelliJ plugin tracker?

**Comment by radhe-soni** [comment]
For me, it mostly happens after a couple of hot reloads or hot restarts.

**Comment by masterwok** [comment]
I've aliased the following command to now nuke all flutter processes. I think it's a little better than restarting the entire IDE.

```
for pid in $(ps -ef | grep "flutter" | awk '{print $2}'); do kill -9 $pid; done
```

**Comment by fbelluco** [comment]
It is being issue for a couple of months already. With VS Code it does not happen, btw.
